src/assetContainer.ts

```diff
@@ -454,4 +458,67 @@ export class AssetContainer extends AbstractScene {
         this.meshes.unshift(rootMesh);
         return rootMesh;
     }
+
+    /**
+     * Merge animations from this asset container into a scene
+     * @param scene is the instance of BABYLON.Scene to append to (default: last created scene)
+     * @param targetConverter defines a function used to convert animation targets from the asset container to the scene (default: search node by name)
+     */
+    public mergeAnimationsTo(scene: Nullable<Scene> = EngineStore.LastCreatedScene, targetConverter: Nullable<(target: any) => Nullable<Node>> = null): void {
+        if (!scene) {
+            Logger.Error("No scene available to merge animations to");
+            return;
+        }
+
+        let _targetConverter = targetConverter ? targetConverter : (target: any) => { return scene.getBoneByName(target.name) || scene.getNodeByName(target.name) };
+
+        // Copy node animations
+        let nodesInAC = this.getNodes();
+        nodesInAC.forEach(nodeInAC => {
+            let nodeInScene = _targetConverter(nodeInAC);
+            if (nodeInScene != null) {
+                // Remove old animations with same target property as a new one
+                for (let animationInAC of nodeInAC.animations) {
+                    // Doing treatment on an array for safety measure
+                    let animationsWithSameProperty = nodeInScene.animations.filter(animationInScene => {
+                        return animationInScene.targetProperty === animationInAC.targetProperty
+                    });
+                    for (let animationWithSameProperty of animationsWithSameProperty) {
+                        const index = nodeInScene.animations.indexOf(animationWithSameProperty, 0);
+                        if (index > -1) {
+                            nodeInScene.animations.splice(index, 1);
+                        }
+                    }
+                }
+
+                // Append new animations
+                nodeInScene.animations = nodeInScene.animations.concat(nodeInAC.animations);
+            }
+        });
+
+        // Copy animation groups
+        this.animationGroups.slice().forEach(animationGroupInAC => {
+            // Clone the animation group and all its animatables
+            animationGroupInAC.clone(animationGroupInAC.name, _targetConverter);
+
+            // Remove animatables related to the asset container
+            animationGroupInAC.animatables.forEach(animatable => {
+                animatable.stop();
+            })
+        });
+
+        // Copy animatables
+        scene.animatables.slice().forEach(animatable => {

```

Comment:
Shouldn't that just happen for NEW animations and not all (including the ones already there) ?

Reply:
The next if statement is making sure to hit only new animations

`if (target && target !== animatable.target) {`

Reply:
Well I would argue that this will not be required if you used imported animatables list instead of the entire scene.animatables

Reply:
New animatables from assetContainer are stored directly into the scene.animatables array when autostarting node animations.
I can store the array before that and compare with updated one after animation file import.
The current scene.animatables array would be a parameter of the MergeAnimationsTo method.
Or use the array length as index for the new animatables.

Reply:
done